tests/containers/volumes.pm

```diff
@@ -0,0 +1,125 @@
+    # The --mount flag does not support z or Z options for modifying selinux labels.
+    my $Z = $runtime eq "podman" ? ",Z" : "";
+
+    my $test_dir = "test_dir";
+    my $test_file = "test_file";
+    my $test_image = "test_image";
+    my $test_container = "test_container";
+    my $test_volume = "test_volume";
+
+    assert_script_run("mkdir $test_dir");
+
+    # Create Dockerfile with VOLUME defined
+    assert_script_run("echo -e 'FROM busybox\nVOLUME /$test_dir' > $test_dir/Dockerfile");
+
+    # Build image
+    assert_script_run("$runtime build -t $test_image -f $test_dir/Dockerfile $test_dir/");
+
+    # Test --volumes-from
+    # Case 1: Check that the volume from container is visible in another container
+    assert_script_run("$runtime run -d --name $test_container $test_image");
+    assert_script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir");
+    assert_script_run("$runtime rm -vf $test_container");
+
+    # Case 2: Check that the volume from container is visible in another container, but the
+    # first container is mounting it in the directory specified as VOLUME in the Dockerfile
+    assert_script_run("touch $test_dir/$test_file");
+    assert_script_run("$runtime run -d --name $test_container -v \$PWD/$test_dir:/$test_dir:Z $test_image");
+    # NOTE: When https://github.com/containers/podman/issues/19529 is solved we must add a version check here
+    my $ret = script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir/$test_file");
+    if ($ret != 0) {
+        if ($runtime eq "podman") {
+            record_soft_failure("gh#19529 - Unexpected error with --volumes-from") if ($ret != 0 && $runtime == "podman");
+        } else {
+            die("--volumes-from failed");
+        }
+    }
+
+    assert_script_run("$runtime rm -vf $test_container");
+
+    # Test --volume option with directory (read-only)
+    assert_script_run("! $runtime run --rm --volume \$PWD/$test_dir:/$test_dir:ro,Z $test_image rm /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    # Equivalent --mount option to above
+    assert_script_run("! $runtime run --rm --mount type=bind,source=\$PWD/$test_dir,destination=/$test_dir,readonly$Z $test_image rm /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    assert_script_run("rm $test_dir/$test_file");
+
+    # Test --volume option with directory (read-write)
+    assert_script_run("$runtime run --rm --volume \$PWD/$test_dir:/$test_dir:Z $test_image touch /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    assert_script_run("rm $test_dir/$test_file");
+
+    # Equivalent --mount option to above
+    assert_script_run("$runtime run --rm --mount type=bind,source=\$PWD/$test_dir,destination=/${test_dir}$Z $test_image touch /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    # Test volume subcommands
```

Comment:
Let's move those up and start with the simple volume commands before building a container and diving into the more complex scenarios.

Easy things first.

Reply:
I wouldn't consider the volume subcommands the easy part...

Reply:
After looking at it a second time, I tend to agree.